### PR TITLE
Fix to bug on default purge setting not saving.

### PR DESCRIPTION
**Pull request by madsnorgaard**
Bug fix to enable override of default purge time period.

### CODE REVIEW

File: os2forms_forloeb.module [hunk 1]
```diff
@@ -128,13 +128,19 @@ function end_notification_batch_function($processID, $queueID) {
...
 function os2forms_forloeb_webform_create(\Drupal\webform\WebformInterface $webform) {
   // Set purge of all users submissions.
   $webform->setSetting('purge', 'all');
   // Set purge of submissions more than 30 days old.
-  $webform->setSetting('purge_days', '30');
+  if (empty($webform->getSetting('purge_days'))) {
+    $webform->setSetting('purge_days', '30');
+  }
+  // Save if time period has been changed.
+  if ($save_webform) {
```
Comment (lats1) [inline os2forms_forloeb.module:141]: Where does the `$save_webform` variable come from? I don't think it's defined or set anywhere. On another node, the convention for variable naming in php projects is to use camel case.

**Review by lats1 — CHANGES_REQUESTED** [review]

File: os2forms_forloeb.module (hunk 1 above)
Comment (madsnorgaard) [reply os2forms_forloeb.module]: Point taken and we have added a bit more extensive logic to this feature. 

In terms of naming standards in Drupal and PHP this should be the documentation - [Documentation on naming standards for variables:](https://www.drupal.org/docs/develop/standards/coding-standards#naming) 

_Variables should be named using lowercase, and words should be separated either with uppercase characters (example: $lowerCamelCase) or with an underscore (example: $snake_case). Be consistent; do not mix camelCase and snake_case variable naming inside a file._

File: os2forms_forloeb.module [hunk 2]
```diff
@@ -128,13 +128,31 @@ function end_notification_batch_function($processID, $queueID) {
...
+ */
+function os2forms_forloeb_webform_presave(\Drupal\webform\WebformInterface $webform) {
+    // Add a purge time frame if not set.
+    if (empty($webform->getSetting('purge_days'))) {
+        $webform->setSetting('purge_days', 30);
+        $save_webform = TRUE;
+        // Save if anything changed.
+        if ($save_webform) {
+            $webform->save();
+        }
```
Comment (lats1) [inline os2forms_forloeb.module:154]: As I see it, this is equal to
```
if (TRUE) {
    $webform->save();
}
```
So the webform will always be saved at this point.
Is there a situation where `$save_webform` would have another value than `TRUE`? If so there should probably be some logic that changes the value of `$save_webform`. If the webform always should be saved at this point the check should probably be omitted.

On another point, could the call to the `$webform->save()`  method go wrong?

**Review by lats1 — CHANGES_REQUESTED** [review]

File: os2forms_forloeb.module (hunk 2 above)
Comment (madsnorgaard) [reply os2forms_forloeb.module]: Good point Lars. I have removed the check and it is more minimal and will always save changes.

**Review by lats1 — APPROVED** [review]

**Pull request merged**